lib/action_client/submission_job.rb

```diff
@@ -0,0 +1,33 @@
+module ActionClient
+  class SubmissionJob < ActiveJob::Base
+    def self.after_perform(with_status: nil, **options, &block)
+      if with_status.present?
+        super(
+          if: ->() {
+            status, * = @response
```

Comment:
Is there a name that could emphasise that this is the real response status, rather than the one we're expecting? Between `status`, `with_status` and `http_statuses` I found myself getting a little lost.

Maybe `response_status` instead of `status`, and `expected_statuses` instead of `http_statuses`?

Reply:
I've re-structured this to use `HttpStatusFilter`: https://github.com/thoughtbot/action_client/pull/5/commits/09d3479e3732c69bbdb31225fa70f6f9bc548ddf